core/state/reader.go

```diff
@@ -207,16 +215,39 @@ type trieReader struct {
 
 // trieReader constructs a trie reader of the specific state. An error will be
 // returned if the associated trie specified by root is not existent.
-func newTrieReader(root common.Hash, db *triedb.Database, cache *utils.PointCache) (*trieReader, error) {
+func newTrieReader(root common.Hash, db, verkledb *triedb.Database, cache *utils.PointCache) (*trieReader, error) {
 	var (
-		tr  Trie
-		err error
+		tr   Trie
+		mptr *trie.StateTrie
+		vktr *trie.VerkleTrie
+		err  error
 	)
-	if !db.IsVerkle() {
-		tr, err = trie.NewStateTrie(trie.StateTrieID(root), db)
+	// get the transition status from the MPT triedb
+	ts, err := db.LoadTransitionState(root)
+	if err != nil {
+		return nil, err
+	}
+
+	if !ts.Started {
+		mptr, err = trie.NewStateTrie(trie.StateTrieID(root), db)
+		if err != nil {
+			return nil, err
+		}
+		vktr, err = trie.NewVerkleTrie(common.Hash{}, verkledb, cache)
 	} else {
-		tr, err = trie.NewVerkleTrie(root, db, cache)
+		vktr, err := trie.NewVerkleTrie(root, verkledb, cache)
+		if err != nil {
+			return nil, err
+		}
+		tr = vktr
+		if !ts.Ended {
+			mptr, err = trie.NewStateTrie(trie.StateTrieID(ts.BaseRoot), db)
+			if err != nil {
+				return nil, err
+			}
+		}
 	}
+	tr = trie.NewTransitionTree(mptr, vktr, false)
```

Comment:
so as a follow-up to the message above, this would be `NewTransitionTree(mptr, vktr, false, ts)`